### PR TITLE
Draft on DataLad siblings

### PR DESCRIPTION
**Pull request by adswa**
This adds a basic example of how to add a sibling to a dataset and update from it.
It's meant to be included after #64 and #72.

### CODE REVIEW

File: docs/basics/101-125-siblings.rst [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+.. runrecord:: _examples/DL-101-125-101
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   # navigate into the installed copy
+   $ cd ../mock_user/DataLad-101
+
+   # download the shell script and save it in your code/ directory
+   $ wget -nv https://raw.githubusercontent.com/datalad/datalad.org/7e8e39b1f08d0a54ab521586f27ee918b4441d69/content/asciicast/seamless_nested_repos.sh -O code/nested_repos.sh
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: I think it is better to stick to DataLad commands:

```
datalad download-url \
  -m "Include nesting demo from datalad website" \
  -O code/nested_repos.sh \ 
  https://raw.githubusercontent.com/datalad/datalad.org/7e8e39b1f08d0a54ab521586f27ee918b4441d69/content/asciicast/seamless_nested_repos.sh
```

File: docs/basics/101-125-siblings.rst [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+   $ datalad status
+
+Save this file:
+
+.. runrecord:: _examples/DL-101-125-103
+   :language: console
+   :workdir: dl-101/mock_user/DataLad-101
+
+   $ datalad save -m "add shell script from ascii cast on nesting" code/nested_repos.sh
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: With `download-url` everything until here becomes obsolete.

File: docs/basics/101-125-siblings.rst [hunk 3]
```diff
@@ -0,0 +1,190 @@
+Dataset siblings
+----------------
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: Following #87 this should likely be talking about "collaboration"

File: docs/basics/101-125-siblings.rst [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+So how do we link back from the copy of the dataset to its
+origin, such that your room mates changes can be included in
+your dataset? How to we let the original dataset "know" about
+this copy your room mate has?
+Do we need to install the installed dataset of our room mate
+as a copy again?
+
+No, luckily, it's simpler and less convoluted. What we have to
+do is to create a datalad ``sibling``: A reference to our room mates
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: s/create/register/ -- it is already there, just needs to be announced.

File: docs/basics/101-125-siblings.rst [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+dataset in our own, original dataset.
+
+.. gitusernote::
+
+   Git repositories can configure clones of a dataset as *remotes* in
+   order to fetch, pull, or push from and to them. A ``datalad sibling``
+   is the equivalent of a git clone that is configured as a remote.
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: Rational for using a term other than *remote*: Git operates on *remotes*, git annex adds the concept of *special remotes*. For DataLad I felt that we needed a term that is a superset of both. "special remote" can only be understood, if the meaning/purpose/scope of a "non-special remote" is understood.

In DataLad we want to do nifty things like making a non-git-annex regular remote on github behave like as if it had an annex, by configuring a companion special remote (see https://github.com/datalad/datalad/issues/335, which is the way the studyforrest dataset is distributed).

Having to distinguish these technicalities all the time would be needlessly complex, IMHO.

File: docs/basics/101-125-siblings.rst [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+.. runrecord:: _examples/DL-101-125-104
+   :language: console
+   :workdir: dl-101/mock_user/DataLad-101
+
+   cd ../../DataLad-101
+   # add a sibling
+   $ datalad siblings add -d . --name roommate --url ../mock_user/DataLad-101
+
+There are a few confusing parts about this command: For one, don't be surprised
+about the ``--url`` argument -- it's called "URL" but it can be a path as well.
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: FTR: "url" is what `git remote add` documentation uses too.

File: docs/basics/101-125-siblings.rst [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+Also, don't forget to give a name to your datasets sibling. Without the ``-s``/
+``--name`` argument the command will fail. The reason behind this is that the default
+name of a sibling if no name is given will be the hostname of the specified URL,
+but as you provide a path and not a URL, there is no hostname to take as a default.
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: Good catch, but really is a bug that should be fixed: https://github.com/datalad/datalad/issues/3553

File: docs/basics/101-125-siblings.rst [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad update --merge -s roommate
+
+Note that we supplied the siblings name with the ``-s``/``--name`` option.
+This is good practice, and allows you to be precise in where you want to get
+updates from. It would have worked without the specification (just as a bare
+``datalad update --merge`` worked for your room mate), because there is only
+one known other location, though.
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: s/one known other/one other known/

File: docs/basics/101-125-siblings.rst [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+in the resulting list with the name "roommate" you have given to it.
+
+The fact that the ``DataLad-101`` dataset now has a sibling means that we
+can also ``datalad update`` this repository:
+
+.. runrecord:: _examples/DL-101-125-106
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad update --merge -s roommate
```
Comment (mih) [inline docs/basics/101-125-siblings.rst]: Consider not going for the `--merge` right away, but plain `update` first, then `datalad diff` against the remote, see the coming change, then merge.

**Comment by adswa** [comment]
Thx for the comments! I've included your suggestions, but updating the examples is yet a TODO and will be much easier than currently when #64 is merged.

**Comment by adswa** [comment]
I've worked in your comments and adjusted filenames to finally have subsequent numbering again. I'll merge once tests pass.

**Pull request merged**